docs/stdlib/datetime.rst

```diff
@@ -612,25 +612,27 @@ EdgeDB stores and outputs timezone-aware values in UTC.
         ...   <duration>'2 hours';
         {-3600s}
 
-    It is an error to subtract a date/time object from a time interval:
+    When subtracting a date/time object from a time interval, an exception
```

Comment:
Let's restructure this section a bit. We can add a note block that introduces the idea that some combinations of values do not make sense to subtract from one another. Within that paragraph, we can individually introduce each incompatible combination followed by the REPL example showing the result of attempting that subtraction.

I would offer a proper GitHub suggestion here, but it's a bit of a nightmare trying to line everything up and edit line-by-line here in the comment box. Intro might be something like this:

> Subtraction doesn't make sense for some type combinations. You couldn't subtract a point in time from a duration, so neither can EdgeDB (although the inverse — subtracting a duration from a point in time — is perfectly fine). You also couldn't subtract a timezone-aware datetime from a local one or vice versa. If you attempt any of these, EdgeDB will raise an exception as shown in these examples:

The note about subtracting `local_date` values should be a separate note block (or at least separate _from_ the incompatible subtraction note block).